distribution/pull_v2_windows.go

```diff
@@ -91,8 +90,8 @@ func filterManifests(manifests []manifestlist.ManifestDescriptor, p ocispec.Plat
 			continue
 		}
 		// TODO(thaJeztah): should we also take the user-provided platform into account (if any)?
-		if strings.EqualFold("windows", manifestDescriptor.Platform.OS) {
-			if err := checkImageCompatibility("windows", manifestDescriptor.Platform.OSVersion); err != nil {
+		if manifestDescriptor.Platform.OS == "windows" {
```

Comment:
This check is repeated inside `checkImageCompatibility`. (I'd personally also rename that function to `checkImageOSVersion` or something so that it's super-clear what it's checking and not-checking from its inputs.